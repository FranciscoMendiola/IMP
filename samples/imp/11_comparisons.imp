if a = b then
  skip
else
  if x <= y then
    z := 1
  else
    z := 0
