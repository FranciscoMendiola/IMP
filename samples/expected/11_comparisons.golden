Token {tokenKind = KIf, lexeme = "if", position = 0}
Token {tokenKind = Id, lexeme = "a", position = 3}
Token {tokenKind = Eq, lexeme = "=", position = 5}
Token {tokenKind = Id, lexeme = "b", position = 7}
Token {tokenKind = KThen, lexeme = "then", position = 9}
Token {tokenKind = KSkip, lexeme = "skip", position = 16}
Token {tokenKind = KElse, lexeme = "else", position = 21}
Token {tokenKind = KIf, lexeme = "if", position = 28}
Token {tokenKind = Id, lexeme = "x", position = 31}
Token {tokenKind = Leq, lexeme = "<=", position = 33}
Token {tokenKind = Id, lexeme = "y", position = 36}
Token {tokenKind = KThen, lexeme = "then", position = 38}
Token {tokenKind = Id, lexeme = "z", position = 47}
Token {tokenKind = Assign, lexeme = ":=", position = 49}
Token {tokenKind = IntLit, lexeme = "1", position = 52}
Token {tokenKind = KElse, lexeme = "else", position = 56}
Token {tokenKind = Id, lexeme = "z", position = 65}
Token {tokenKind = Assign, lexeme = ":=", position = 67}
Token {tokenKind = IntLit, lexeme = "0", position = 70}
